cwms-data-api/src/main/java/cwms/cda/api/location/kind/BaseOutletHandler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * MIT License
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.api.location.kind;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.api.Controllers;
+import io.javalin.http.Handler;
+import static com.codahale.metrics.MetricRegistry.name;
+import static cwms.cda.api.Controllers.RESULTS;
+import static cwms.cda.api.Controllers.SIZE;
+
+abstract class BaseOutletHandler implements Handler {
```

Comment:
True, it's more of a BaseHandler class.  I wasn't sure if I should introduce it as an abstract class everyone can use or not, because it seemed like the norm was to write it all out.  I can move it to the root package and make it less specific if you want.

Reply:
Definitely a good way to start, but this definitely looks like something they all use. Though probably reasonable to keep the specific name until someone wants to go through refactor all of them.

Reply:
@adamkorynta @MikeNeilson is this something that needs to be resolved?

Reply:
Only if you really want to slog through all usages of `Handler.`

Reply:
Handler is pretty small, but CrudHandler is the big lift. I think the BaseHandler class shouldn't implement an interface that way both Handler/CrudHandler impls can use the same logic. I'd do all that in a separate PR though.

Reply:
If we do a base class agnostic of the Handler and CrudHandler interface, that's 51 classes that'll need an update.  I'm down for doing that in a separate PR, but where does that time go?  We can always create a ticket, and put that on hold until we can decide on that?

Reply:
I would just create the ticket.

Reply:
It'll either get approved, or one of us will just get that annoyed when we see it again.

Reply:
#811 created for this.